src/Butler/Auth/OIDC.hs

```diff
@@ -126,40 +126,42 @@ loginServer sessions mkIndexHtml jwtSettings oidcenv auth mWorkspace =
         }
   where
     callbackRoute :: Maybe Text -> Maybe Text -> Maybe Text -> ServantProcessIO AuthResp
-    callbackRoute mErr mCode mState = do
-        case (mErr, mCode, mState) of
-            (Just errorMsg, _, _) -> error $ "Error from remote provider: " <> show errorMsg
-            (_, Nothing, _) -> error "No code parameter given"
-            (_, _, Nothing) -> error "No state parameter given"
-            (_, Just oauthCode, Just oauthState) -> do
-                tokens :: O.Tokens Value <-
-                    liftIO $
-                        O.getValidTokens
-                            (mkSessionStore oidcenv (Just $ encodeUtf8 oauthState) Nothing)
-                            (oidc oidcenv)
-                            (manager oidcenv)
-                            (encodeUtf8 oauthState)
-                            (encodeUtf8 oauthCode)
-                let username = UserName tokens.idToken.sub
-                setSession username $ externalProvider "google" username
+    callbackRoute mErr mCode mState = withSession \case
+        Nothing -> guestRedirect -- this is weird, the user should already have a guest session at this point.
```

Comment:
I just tried the PR and it looks like I'm in that case after my social login I'm redirected and I'm logged as guest :( Each time I hit the login button I a new guest.

Reply:
Could you check if there was a cookie in the request header? I wonder how can that happens, perhaps you went straight to the "/_login" endpoint which doesn't set a cookie. Then I guess it should be safe to create a session instead of doing the redirection here.